src/components/molecules/Signup.tsx

```diff
@@ -0,0 +1,342 @@
+});
+
+const StyledIndicator = styled(CheckboxPrimitive.Indicator, {
+  color: indigo.indigo11,
+});
+
+// Exports
+export const TabsContent = StyledContent;
+export const Checkbox = StyledCheckbox;
+export const CheckboxIndicator = StyledIndicator;
+
+const Flex = styled("div", { display: "flex" });
+
+const Text = styled("div", {
+  marginBottom: 20,
+  color: mauve.mauve11,
+  fontSize: 15,
+  lineHeight: 1.5,
+});
+
+const Annotation = styled("p", {
+  marginTop: 5,
+  color: mauve.mauve11,
+  fontSize: 12,
+  lineHeight: 1.5,
+});
+
+const Button = styled("button", {
+  all: "unset",
+  display: "inline-flex",
+  alignItems: "center",
+  justifyContent: "center",
+  borderRadius: 4,
+  padding: "0 15px",
+  fontSize: 15,
+  lineHeight: 1,
+  fontWeight: 500,
+  height: 35,
+
+  variants: {
+    variant: {
+      indigo: {
+        backgroundColor: indigo.indigo4,
+        color: indigo.indigo11,
+        "&:hover": { backgroundColor: indigo.indigo5 },
+        "&:focus": { boxShadow: `0 0 0 2px ${indigo.indigo7}` },
+      },
+    },
+  },
+
+  defaultVariants: {
+    variant: "indigo",
+  },
+});
+const Fieldset = styled("fieldset", {
+  all: "unset",
+  marginBottom: 15,
+  width: "100%",
+  display: "flex",
+  flexDirection: "column",
+  justifyContent: "flex-start",
+});
+
+const Label = styled("label", {
+  fontSize: 13,
+  lineHeight: 1,
+  marginBottom: 10,
+  color: indigo.indigo12,
+  display: "block",
+});
+
+const Input = styled("input", {
+  all: "unset",
+  flex: "1 0 auto",
+  borderRadius: 4,
+  padding: "0 10px",
+  fontSize: 15,
+  lineHeight: 1,
+  color: indigo.indigo11,
+  boxShadow: `0 0 0 1px ${indigo.indigo7}`,
+  height: 35,
+  "&:focus": { boxShadow: `0 0 0 2px ${indigo.indigo8}` },
+  "&::placeholder": { color: mauve.mauve7 },
+});
+
+const FieldCheck = styled("fieldset", {
+  all: "unset",
+  width: "100%",
+  display: "flex",
+  alignItems: "center",
+});
+
+const CheckLink = styled("a", {
+  color: indigo.indigo11,
+  textDecoration: "underline",
+  "&:visited": { color: indigo.indigo11 },
+});
+
+const ErrorMsg = styled("p", {
+  color: tomato.tomato8,
+  fontSize: 12,
+  marginTop: 5,
+});
+
+export default function Form({ setLoggedin }: Props) {
+  const {
+    register,
+    handleSubmit,
+    formState: { errors },
+    getValues,
+  } = useForm<SignupFormInput>();
+  const [isChecked, setChecked] = useState<boolean>(false);
+  const [errMessage, setErrMessage] = useState<string | null>();
+
+  const signupOnSubmit: SubmitHandler<SignupFormInput> = (data) => {
+    setErrMessage("");
+    const sendData = JSON.stringify({
+      first_name: data.firstname,
+      last_name: data.lastname,
+      email: data.email,
+      phone_number: data.phone,
+      zip_code: data.zipcode,
+      serial_number: data.serialnumber,
+      password: data.password,
+    });
+
+    const customConfig = {
+      headers: {
+        "Content-Type": "application/json",
+      },
+    };
```

Comment:
Do you need this config?
I think Axios will automatically handle json.

Reply:
After trying and error, this description seems to be necessary.